recipes/IEMOCAP/emotion_recognition/train_with_wav2vev2.py

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env python3
+"""Recipe for training an emotion recognition system from speech data only using IEMOCAP.
+The system classifies 4 emotions ( anger, happiness, sadness, neutrality) with wav2vec2.
+
+To run this recipe, do the following:
+> python train_with_wav2vec2.py hparams/train_with_wav2vec2.yaml --data_folder /path/to/IEMOCAP
+
+Authors
+ * Yingzhi WANG 2021
+"""
+
+import os
+import sys
+import speechbrain as sb
+import torch
+from hyperpyyaml import load_hyperpyyaml
+
+
+class EmoIdBrain(sb.Brain):
+    def compute_forward(self, batch, stage):
+        """Computation pipeline based on a encoder + emotion classifier.
+        """
+        batch = batch.to(self.device)
+        wavs, lens = batch.sig
+
+        wav2vec2_out = self.modules.wav2vec2(wavs)
+
+        average_pooling = sb.nnet.pooling.Pooling1d(
+            pool_type="avg",
+            input_dims=3,
+            kernel_size=wav2vec2_out.shape[1],
+            pool_axis=1,
+        )
+
+        averaged_out = average_pooling(wav2vec2_out).view(
+            wav2vec2_out.shape[0], -1
+        )
+
+        outputs = self.modules.output_mlp(averaged_out)
+        outputs = self.hparams.log_softmax(outputs)
+        return outputs
+
+    def compute_objectives(self, predictions, batch, stage):
+        """Computes the loss using speaker-id as label.
+        """
+        emoid, _ = batch.to(self.device).emo_encoded
```

Comment:
Why loading the batch with the specific device in your data_io function ?

Reply:
Done